### PR TITLE
src: fix lint issue with options

### PR DESCRIPTION
**Pull request by helio-frota**
Able to fix some lint issues by using TS approach described here: https://github.com/yargs/yargs/blob/master/docs/typescript.md#typescript-usage-examples

Also removed `fields` and `organization` that are not part of top level options.

### CODE REVIEW

**Comment by helio-frota** [comment]
result:

```console
$ node build/cli.js 
cli.js <command>

Commands:
  cli.js licenses  Searches for licenses of the project.

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --cwd      The current working directory
```

**Comment by helio-frota** [comment]
Also suppressing the list warning related to the result that comes from license-checker because we worked on the result content by transforming that in a `LicenseData` type.

File: src/util/githubFetcher.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios';
 
 // To avoid throwing errors with 404.
-axios.defaults.validateStatus = (_) => true;
+axios.defaults.validateStatus = function () { return true };
```
Comment (alexalikiotis) [inline src/util/githubFetcher.ts:4]: @helio-frota I do think you can just write `axios.defaults.validateStatus = () => true;`

I'm not a big fun of the `= function () { ... }` syntax cause it feels wired

If you have a function expression just using an arrow function I think it's the best approach

Comment (helio-frota) [reply src/util/githubFetcher.ts:4]: if I'm not wrong that is causing lint issues also.. but I'll look again

Comment (helio-frota) [reply src/util/githubFetcher.ts]: thank you @alexalikiotis , suggestion added, code updated :+1:

**Review by alexalikiotis — APPROVED** [review]
LGTM

**Pull request merged**